feat: new method `create_operator_directed_avs_rewards_submission`

damiramirez opened the pull request:
### What Changed?
Add new method `create_operator_directed_avs_rewards_submission` to `avsregistry/writer`.

Related issue https://github.com/Layr-Labs/eigensdk-rs/issues/312

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

damiramirez committed:
WIP: Test failing for CALCULATION_INTERVAL_SECONDS

damiramirez committed:
Fix test

damiramirez committed:
Improve test case

damiramirez committed:
Update CHANGELOG.md with example

damiramirez committed:
Add a comment to the URL of the contract

damiramirez committed:
Add docs to method

damiramirez committed:
Remove solidity comments

damiramirez committed:
Add more context to the duration and start timestamp values

damiramirez committed:
Use IRegistryCoordinator to get values instead of literals

damiramirez committed:
Merge branch 'dev' into feat/create-operator-avs-rewards

MegaRedHand commented on crates/chainio/clients/avsregistry/src/writer.rs:
```diff
@@ -681,6 +714,91 @@ mod tests {
+            .await
+            .unwrap();
+
+        assert!(is_registered,);
```
```suggestion
        assert!(is_registered);
```

damiramirez committed:
Update crates/chainio/clients/avsregistry/src/writer.rs

Co-authored-by: Tomás Grüner <47506558+MegaRedHand@users.noreply.github.com>

damiramirez committed:
Fix link to contracts

MegaRedHand committed:
Merge branch 'dev' into feat/create-operator-avs-rewards

MegaRedHand committed:
Merge branch 'dev' into feat/create-operator-avs-rewards

MegaRedHand committed:
fix: remove GENESIS_TIMESTAMP constant

We can't depend on the constant since any state dump regeneration would make it stale.

MegaRedHand committed:
Merge branch 'dev' into feat/create-operator-avs-rewards

damiramirez committed:
Merge branch 'dev' into feat/create-operator-avs-rewards
